train.py

```diff
@@ -244,7 +244,12 @@ def loss_fn(pred, labels):
     optimizer = build_optimizer(model, job_config)
     scheduler = get_lr_scheduler(optimizer, job_config)
 
-    metric_logger = build_metric_logger(job_config)
+    if parallel_dims.pp_enabled:
+        pp_size = pp_mesh.size()
+        metrics_log_rank = int((world_mesh.size() // pp_size) * (pp_size - 1))
```

Comment:
Does this rank computation assume that PP is outermost? If so, should we assert/check for that?

Reply:
yea, it does. I wonder if there is a better place to do the assert.  I'll add it here for now.

I also don't like to do it this way, i might want to propose adding a device-mesh API to help do a calculation like this more robustly.

Reply:
i moved this into a util and added an assert.